### PR TITLE
Improve app SYNOPSIS on empty engine_synopsis; align CMP app option order

### PR DESCRIPTION
**Pull request by DDvO**
This is a follow-up on https://github.com/openssl/openssl/pull/12924#issuecomment-695179026.

I found that explicit and implicit empty lines within the SYNOPSIS section of POD files cause layout errors
because they are (mis-)interpreted as the start of a new command (with its own options).

This causes trouble in all apps having `-engine` options, which are disabled by default.
As a workaround I removed the newline after each occurrence of 
`{- $OpenSSL::safe::opt_engine_synopsis -}`

I would have liked to structure the large list of CMP app options by having a newline or even better an intermediate header before each group of options, but apparently there is no possibility to influence the layout within a SYNOPSIS section, not even by using escape/control symbols (which are not recognized for some reason).
I nevertheless added some sort of header texts and would be inclined to accept the sub-optimal layout there.

Moreover, I improved the alignment of the order of the CMP app options.

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man1/openssl-cmp.pod.in
Comment (levitte) [inline doc/man1/openssl-cmp.pod.in:4]: Extra empty line

File: doc/man1/openssl-cmp.pod.in
Comment (levitte) [inline doc/man1/openssl-cmp.pod.in:8]: Extra empty line

File: doc/man1/openssl-cmp.pod.in
Comment (levitte) [inline doc/man1/openssl-cmp.pod.in:171]: Extra empty line

File: doc/man1/openssl-cmp.pod.in
Comment (levitte) [inline doc/man1/openssl-cmp.pod.in:263]: Two empty lines?

File: doc/man1/openssl-cmp.pod.in
Comment (levitte) [inline doc/man1/openssl-cmp.pod.in:415]: Two empty lines

File: doc/man1/openssl-cmp.pod.in
Comment (levitte) [inline doc/man1/openssl-cmp.pod.in:760]: Two empty lines?

**Review by levitte — COMMENTED** [review]
Hmmm, are two double empty line deliberate?

File: doc/man1/openssl-cmp.pod.in
Comment (DDvO) [reply doc/man1/openssl-cmp.pod.in]: Yeah, so far before (sub-)section headers there were partly one, partly two empty lines.
At first in this PR I made this consistent in the direction of two empty lines,
but according to your comments I've just replaced all double empty lines by single ones.

**Comment by DDvO** [comment]
Can these doc layout improevments be approved now?

**Comment by DDvO** [comment]
@levitte, is this ok now?

**Comment by DDvO** [comment]
I've just added improvements on the `-verify_email`, `-verify_hostname`, and `-verify_ip` options.

**Comment by DDvO** [comment]
BTW, I wonder why `./util/find-doc-nits -c` is not part of `make doc-nits` ?

**Comment by DDvO** [comment]
Also this one is waiting for approval.

**Comment by DDvO** [comment]
@levitte, since you already had given comments on this two weeks ago,
can you please have a look again and hopefully approve?

**Comment by DDvO** [comment]
Thanks @levitte for your comment on #12934:
> I thought I had already done that... too many other distractions, I guess

This PR here is likely a further such case.

**Comment by richsalz** [comment]
> BTW, I wonder why `./util/find-doc-nits -c` is not part of `make doc-nits` ?

Because doc-nits doesn't require the app and can be run at any time, but the `-c` flag does needs the executable to be built.  So it's a separate target, `cmd-nits`.

**Comment by DDvO** [comment]
Again, @openssl/otc, can this be approved now?
@levitte, all your comments have been addressed since Sep 23.

**Review by mattcaswell — APPROVED** [review]
LGTM

**Comment by DDvO** [comment]
Merged - thanks @mattcaswell for approving

**Pull request closed without merging**
